public/app/plugins/datasource/grafana/components/AnnotationQueryEditor.tsx

```diff
@@ -0,0 +1,133 @@
+import React from 'react';
+import { SelectableValue } from '@grafana/data';
+import { FieldSet, InlineField, InlineFieldRow, InlineSwitch, Select } from '@grafana/ui';
+import { css } from '@emotion/css';
+
+import { TagFilter } from 'app/core/components/TagFilter/TagFilter';
+import { GrafanaAnnotationQuery, GrafanaAnnotationType, GrafanaQuery } from '../types';
+import { getAnnotationTags } from 'app/features/annotations/api';
+
+const filterTooltipContent = (
+  <ul>
+    <li>
+      Dashboard: This will fetch annotation and alert state changes for whole dashboard and show them only on the event
+      {"'"}s originating panel.
+    </li>
+    <li>Tags: This will fetch any annotation events that match the tags filter.</li>
+  </ul>
+);
+
+const matchTooltipContent =
+  'By default Grafana only shows annotations that match all tags in the query. Enabling this returns annotations that match any of the tags in the query.';
+
+const tagsTooltipContent = (
+  <div>
+    A tag entered here as {"'"}foo{"'"} will match
+    <ul>
+      <li>
+        annotation tags {"'"}foo{"'"}
+      </li>
+      <li>
+        annotation key-value tags formatted as {"'"}foo:bar{"'"}
+      </li>
+    </ul>
+  </div>
+);
+
+const annotationTypes = [
+  { label: 'Dashboard', value: GrafanaAnnotationType.Dashboard },
+  { label: 'Tags', value: GrafanaAnnotationType.Tags },
+];
+
+const limitOptions = [10, 50, 100, 200, 300, 500, 1000, 2000].map((limit) => ({
+  label: String(limit),
+  value: limit,
+}));
+
+interface Props {
+  query: GrafanaQuery;
```

Comment:
really this should be typed as `GrafanaAnnotationQuery`, but doing this creates type errors in `datasource`